sql/src/main/java/org/apache/apex/malhar/sql/SQLExecEnvironment.java

```diff
@@ -0,0 +1,248 @@
+import org.apache.calcite.tools.Frameworks;
+import org.apache.calcite.tools.Planner;
+
+import com.datatorrent.api.DAG;
+
+/**
+ * SQL Execution Environment for SQL integration API of Apex.
+ * This exposes calcite functionality in simple way. Most of the APIs are with builder pattern which makes it
+ * easier to construct a DAG using this object.
+ *
+ * Eg.
+ * <pre>
+ * SQLExecEnvironment.getEnvironment(dag)
+ *                   .registerTable("TABLE1", Object of type {@link Endpoint})
+ *                   .registerTable("TABLE2", Object of type {@link Endpoint})
+ *                   .executeSQL("INSERT INTO TABLE2 SELECT STREAM * FROM TABLE1);
+ * </pre>
+ *
+ * Above code will evaluate SQL statement and convert the resultant Relational Algebra to a sub-DAG.
+ */
+public class SQLExecEnvironment
+{
+  private final JavaTypeFactoryImpl typeFactory;
+  private SchemaPlus schema = Frameworks.createRootSchema(true);
+  private DAG dag;
+
+  /**
+   * Construct SQL Execution Environment which works on given DAG objec.
+   *
+   * @param dag LogicalPlan object that will be populated with sub-DAG for SQL operations.
+   */
+  private SQLExecEnvironment(DAG dag)
+  {
+    this.dag = dag;
+    this.typeFactory = new JavaTypeFactoryImpl(RelDataTypeSystem.DEFAULT);
+  }
+
+  /**
+   * Given SQLExec{@link SQLExecEnvironment} object for given {@link DAG}.
+   *
+   * @param dag Object of type {@link DAG}
+   * @return Returns {@link SQLExecEnvironment} object
+   */
+  public static SQLExecEnvironment getEnvironment(DAG dag)
+  {
+    return new SQLExecEnvironment(dag);
+  }
+
+  /**
+   * Use given model file to initialize {@link SQLExecEnvironment}.
+   * The model file contains definitions of endpoints and data formats.
+   * Example of file format is like following:
+   * <pre>
+   *   {
+   *     version: '1.0',
+   *     defaultSchema: 'APEX',
+   *     schemas: [{
+   *       name: 'APEX',
+   *       tables: [
+   *         {
+   *            name: 'ORDERS',
+   *            type: 'custom',
+   *            factory: 'org.apache.apex.malhar.sql.schema.ApexSQLTableFactory',
+   *            stream: {
+   *              stream: true
+   *            },
+   *            operand: {
+   *              endpoint: 'file',
+   *              messageFormat: 'csv',
+   *              endpointOperands: {
+   *                directory: '/tmp/input'
+   *              },
+   *              messageFormatOperands: {
+   *                schema: '{"separator":",","quoteChar":"\\"","fields":[{"name":"RowTime","type":"Date","constraints":{"format":"dd/MM/yyyy hh:mm:ss"}},{"name":"id","type":"Integer"},{"name":"Product","type":"String"},{"name":"units","type":"Integer"}]}'
+   *            }
+   *            }
+   *         }
+   *       ]
+   *     }]
+   *   }
+   * </pre>
+   *
+   * @param model String content of model file.
+   * @return Returns this {@link SQLExecEnvironment}
+   */
+  public SQLExecEnvironment withModel(String model)
```

Comment:
I think at this level its best to keep it as String. I don't think it should be the responsibility of this class to read a file.